core/camel-xml-jaxp/src/main/java/org/apache/camel/converter/jaxp/StaxConverter.java

```diff
@@ -316,7 +319,10 @@ private XMLOutputFactory getXMLOutputFactory() {
 
     private void returnXMLOutputFactory(XMLOutputFactory factory) {
         if (factory != outputFactory) {
-            OUTPUT_FACTORY_POOL.offer(factory);
+            boolean resultOfOffer = OUTPUT_FACTORY_POOL.offer(factory);
+            if (!resultOfOffer) {
+                LOG.error("factory wasn't added into OUTPUT_FACTORY_POOL");
```

Comment:
After updating to Camel `3.12.0` we started to see these errors in our logs (not sure if it is this change or the one above, maybe both).

Are you sure this is an error? I believe it happens when multiple threads concurrently create more factories that the maximum pool capacity. Or am I missing something?

Reply:
We should trust static analysis tool

Reply:
The static analyis tool found that a return value is ignored. This can of be a problem, but the static analysis tool does not know the context of it happening. If this was logged at debug or info level it would be the same for the static analyisis tool.

Reply:
Fixed by @davsclaus in https://github.com/apache/camel/commit/fb67852b8f5fbb525d7b2ecff71f78d18754080c and https://github.com/apache/camel/commit/fb67852b8f5fbb525d7b2ecff71f78d18754080c Thanks!